galaxy_ng/app/tasks/deletion.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from pulp_container.app import tasks as pulp_container_tasks
+from pulpcore.app.tasks import orphan_cleanup, reclaim_space
+
+log = logging.getLogger(__name__)
+
+
+def delete_container_distribution(instance_ids):
+    """Deletes a container distribution and push repository related."""
+
+    log.info("Running container.general_multi_delete to delete distro and repo")
+    pulp_container_tasks.general_multi_delete(instance_ids=instance_ids)
+
+    log.info("Running orphan_cleanup to delete CollectionVersion objects and artifacts")
+    orphan_cleanup(content_pks=None, orphan_protection_time=0)
```

Comment:
I was unsure if I needed to call `reclaim_space` here after the call to `orphan_cleanup`

Reply:
c/c @ipanova

Reply:
There is no need to call the `reclaim_space`.  The space can be reclaimed on a repo, since you've just removed it there is nothing to reclaim on..
Orphan clean up will clean up everything what does not belong to any of the repo, including content + artifacts.

Reply:
re: orphan_protection_time, willingly you can set it globally https://github.com/pulp/pulpcore/blob/master/pulpcore/app/settings.py#L248 https://github.com/pulp/pulpcore/blob/master/docs/configuration/settings.rst#orphan_protection_time